src/material-experimental/form-field/testing/form-field-harness.ts

```diff
@@ -0,0 +1,226 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ComponentHarness,
+  ComponentHarnessConstructor,
+  HarnessPredicate,
+  TestElement
+} from '@angular/cdk/testing';
+import {MatInputHarness} from '@angular/material-experimental/input/testing';
+import {MatSelectHarness} from '@angular/material-experimental/select/testing';
+import {FormFieldHarnessFilters} from './form-field-harness-filters';
+
+// TODO(devversion): support datepicker harness once developed (COMP-203).
+// Also support chip list harness.
+/** Possible harnesses of controls which can be bound to a form-field. */
+export type FormFieldControlHarness = MatInputHarness|MatSelectHarness;
+
+/**
+ * Base class for custom form-field control harnesses. Harnesses for
+ * custom controls with form-fields need to implement this interface.
+ */
+export interface CustomFormFieldControlHarness extends ComponentHarness {}
```

Comment:
Let's call it `MatFormFieldControlHarness` and have `MatInputHarness` and `MatSelectHarness` extend it as well. It might need to go in its own file & build rule to make that work though.

Reply:
@mmalerba done. I had to introduce another nested entry-point so that default controls which are also referenced in the `MatFormFieldHarness` can import the base control harness without introducing a circular dependency. For public consumers there is no change. They can still import through `form-field/testing`.